prepared_cache.go

```diff
@@ -1,61 +1,43 @@
 package gocql
 
 import (
-	"github.com/gocql/gocql/internal/lru"
-	"sync"
+	"github.com/cornelk/hashmap"
 )
 
 const defaultMaxPreparedStmts = 1000
 
 // preparedLRU is the prepared statement cache
 type preparedLRU struct {
-	mu  sync.Mutex
-	lru *lru.Cache
+	cache *hashmap.HashMap
 }
 
 // Max adjusts the maximum size of the cache and cleans up the oldest records if
 // the new max is lower than the previous value. Not concurrency safe.
 func (p *preparedLRU) max(max int) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
-
-	for p.lru.Len() > max {
-		p.lru.RemoveOldest()
+	if p.cache.Len() > max {
```

Comment:
So now this is not matching the description, as it grows the current cache instead of clearing the old entries. If the implementation is beneficial, I'd update the description to match ;-)

Reply:
Yes I just hacked away. It probably needs more docs updates. :)

Reply:
But won't this mean the cache can grow without limits now? We had a similar issue on Scylla side and discovered that there are workloads that will eventually make your application run out of memory...